invictus_dashboard/src/app/dashboard/components/dashboard-task-edit/dashboard-task-edit.component.ts

```diff
@@ -281,44 +339,77 @@ export class DashboardTaskEditComponent implements OnInit, OnDestroy {
     }
 
     if (subTaskAltered) {
-      console.log('sub-tasks added. updating...');
+      // console.log('sub-tasks added. updating...');
       this.dashboardService.updateTask(task).subscribe({
         next: () => this.onSaveComplete(),
-        error: (err) => (this.apiErrMsg = err),
+        error: (err) => {
+          this.apiErrMsg = err;
+          console.log(err);
+        },
       });
     } else {
       this.onSaveComplete();
     }
   }
 
-  onCancel() {
-    //TODO: flesh out this alert menu with back button (modal?)
-    alert(
-      'Warning: leaving the form will clear unsaved data. Are you sure you want to exit?'
-    );
-    this.taskForm.reset();
+  onCancel(confirm: boolean) {
+    if (confirm) {
+      this.onSaveComplete();
+    }
   }
 
-  onDelete() {
-    //TODO: flesh out the alert for this too
+  onDelete(confirm: boolean) {
+    console.log('delete confirmed: ', confirm);
     if (!this.task.id || this.task.id === 0) {
-      this.onCancel();
+      if (!confirm) {
+        this.onSaveComplete();
+      }
     } else {
-      this.dashboardService.deleteTask(this.task.id).subscribe({
-        next: () => this.onSaveComplete(),
-        error: (err) => (this.apiErrMsg = err),
-      });
+      if (confirm) {
+        this.dashboardService.deleteTask(this.task.id).subscribe({
+          next: () => this.onSaveComplete(),
+          error: (err) => (this.apiErrMsg = err),
+        });
+      }
     }
   }
 
-  onSaveComplete() {
+  setDialogText(isDelete: boolean): string {
+    var modalText: string = '';
+    if (isDelete) {
+      modalText = `Delete ${
+        this.taskForm.get('title').value
+      }? This action cannot be undone.`;
+    } else {
+      modalText =
+        'Leaving the form will clear unsaved data. Are you sure you want to exit?';
+    }
+    return modalText;
+  }
+
+  onSaveComplete(): void {
     this.taskForm.reset();
     this.taskChangeEvent.emit('task');
-    document.getElementById('exitBtn').click();
+    this.exitBtnElement.nativeElement.click();
```

Comment:
We want to avoid DOM manipulation even when it has safe references. Instead of calling click on a button you could do the actions that are within the click. Observables can help you with this and we'll get there soon.